include/hip/devicelib/double_precision/dp_intrinsics.hh

```diff
@@ -20,39 +20,242 @@
  * DEALINGS IN THE SOFTWARE.
  */
 
-
 #ifndef HIP_INCLUDE_DEVICELIB_DP_INTRINSICS_H
 #define HIP_INCLUDE_DEVICELIB_DP_INTRINSICS_H
 
 #include <hip/devicelib/macros.hh>
+#include "CHIPSPVConfig.hh"
+
+/**
+ * @brief Declare as extern - we state that these funcitons are implemented and
+ * will be found at link time
+ *
+ * The format is as follows:
+ * 1. Declare the external function which will be executed with the appropriate
+ * linkage. Inline comment specifying where the implementation is coming from.
+ * (OpenCL, OCML, custom) note: some of these declarations are not strictly
+ * necesary but are included for completeness.
+ * 2. If necessary, define the type specific function and bind it to the
+ * function declared in 1. cosf(x) -> cos(x)
+ */
+
+#if defined(OCML_BASIC_ROUNDED_OPERATIONS)
+// extern "C++" __device__ double __ocml_add_rtn_f64(double x, double y);
```

Comment:
Not really. Ideally, these declarations would live here but some of these OMCL functions are declared in `spirv_math_fwd.h` or all forward declarations should live there. I tried removing this file but there are some dependencies so I'll leave this as is since I can't invest any more time into this right now.